tests/features/capture_output_test.py

```diff
@@ -0,0 +1,88 @@
+import pytest
+
+from pyupgrade._data import Settings
+from pyupgrade._main import _fix_plugins
+
+
+@pytest.mark.parametrize(
+    ('s', 'version'),
+    (
+        pytest.param(
+            'import subprocess\n'
+            'subprocess.run(["foo"], stdout=subprocess.PIPE, '
+            'stderr=subprocess.PIPE)\n',
+            (3,),
+            id='not Python3.7+',
+        ),
+        pytest.param(
+            'from foo import run\n'
+            'import subprocess\n'
+            'run(["foo"], stdout=subprocess.PIPE, stderr=subprocess.PIPE)\n',
+            (3, 7),
+            id='run imported, but not from subprocess',
+        ),
+        pytest.param(
+            'from foo import PIPE\n'
+            'from subprocess import run\n'
+            'subprocess.run(["foo"], stdout=PIPE, stderr=PIPE)\n',
+            (3, 7),
+            id='PIPE imported, but not from subprocess',
+        ),
+        pytest.param(
+            'from subprocess import run\n'
+            'run(["foo"], stdout=None, stderr=PIPE)\n',
+            (3, 7),
+            id='stdout not subprocess.PIPE',
+        ),
+    ),
+)
+def test_fix_capture_output_noop(s, version):
+    assert _fix_plugins(s, settings=Settings(min_version=version)) == s
+
+
+@pytest.mark.parametrize(
+    ('s', 'expected'),
+    (
+        pytest.param(
+            'import subprocess\n'
+            'subprocess.run(["foo"], stdout=subprocess.PIPE, '
+            'stderr=subprocess.PIPE)\n',
+            'import subprocess\n'
+            'subprocess.run(["foo"], capture_output=True)\n',
+            id='subprocess.run and subprocess.PIPE attributes',
+        ),
+        pytest.param(
+            'from subprocess import run, PIPE\n'
+            'run(["foo"], stdout=PIPE, stderr=PIPE)\n',
+            'from subprocess import run, PIPE\n'
+            'run(["foo"], capture_output=True)\n',
+            id='run and PIPE imported from subprocess',
+        ),
+        pytest.param(
+            'from subprocess import run, PIPE\n'
+            'run(["foo"], shell=True, stdout=PIPE, stderr=PIPE)\n',
+            'from subprocess import run, PIPE\n'
+            'run(["foo"], shell=True, capture_output=True)\n',
+            id='other argument used too',
+        ),
+        pytest.param(
+            'import subprocess\n'
+            'subprocess.run(["foo"], stderr=subprocess.PIPE, '
+            'stdout=subprocess.PIPE)\n',
+            'import subprocess\n'
+            'subprocess.run(["foo"], capture_output=True)\n',
+            id='stderr used before stdout',
+        ),
+        pytest.param(
+            'import subprocess\n'
+            'subprocess.run(stderr=subprocess.PIPE, args=["foo"], '
+            'stdout=subprocess.PIPE)\n',
+            'import subprocess\n'
+            'subprocess.run( args=["foo"], capture_output=True)\n',
```

Comment:
this extra space seems wrong

Reply:
This is the same as currently happens for `open(mode='r', file='foo')`, which becomes `open( file='foo')`

I'll add extra logic here and in `open` to remove this leading space if you like, although when I'd done that for `open` you'd commented that it seemed complicated and that this wouldn't come up much in practice (see this review comment: https://github.com/asottile/pyupgrade/pull/412#discussion_r597484470 ). IMO this is the same situtation - it's extremely rare that someone would call `subprocess.run(stderr=subprocess.PIPE, args=["foo"], stdout=subprocess.PIPE)`, and if someone's using pyupgrade they're likely also using a formatter which would take care of extra whitespaces

Reply:
I've tried addressing this in the latest commit anyway